pkg/utils/release/controller.go

```diff
@@ -121,17 +115,160 @@ func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1a
 	return release, err
 }
 
-// Get ReleaseLink object from a given namespace
-func (s *SuiteController) GetReleaseLink(name string, namespace string) (*v1alpha1.ReleaseLink, error) {
-	namespacedName := types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
+// GetReleasePlan returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlanAdmission(releasePlanAdmissionName, devNamespace string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	releasePlanAdmission := &appstudiov1alpha1.ReleasePlanAdmission{}
+
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      releasePlanAdmissionName,
+		Namespace: devNamespace,
+	}, releasePlanAdmission)
+
+	return releasePlanAdmission, err
+}
+
+// DeleteReleasePlan deletes the releasePlan resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlanAdmission(releasePlanAdmissionName, devNamespace string, reportErrorOnNotFound bool) error {
+	releasePlanAdmission := appstudiov1alpha1.ReleasePlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      releasePlanAdmissionName,
+			Namespace: devNamespace,
+		},
+	}
+	err := s.KubeRest().Delete(context.TODO(), &releasePlanAdmission)
+	if err != nil && !reportErrorOnNotFound && k8sErrors.IsNotFound(err) {
+		err = nil
+	}
+	return err
+}
+
+// CreateReleasePlan creates a new ReleasePlan using the given parameters.
+func (s *SuiteController) CreateReleasePlan(name, devNamespace, application, managedNamespace, AutoReleaseLabel string) (*appstudiov1alpha1.ReleasePlan, error) {
```

Comment:
`namespace` and `targetNamespace` are better names. Also, why to capitalize `AutoReleaseLabel`?